controllers/certificatesigningrequest_controller.go

```diff
@@ -0,0 +1,431 @@
+/*
+Copyright 2023 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	"github.com/cert-manager/cert-manager/pkg/controller/certificatesigningrequests/util"
+	"github.com/go-logr/logr"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/clock"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	v1alpha1 "github.com/cert-manager/issuer-lib/api/v1alpha1"
```

Comment:
I had not noticed `v1alpha1.Issuer` before. Why aren't we using the types that already exist in `github.com/cert-manager/pkg/apis`? Or do they not exist?

If we need these `IssuerConditionReason` constants, why do we use a version in the type? I though these versions only made sense in the context of CRDs, but we aren't defining the types for CRDs right?

**Context:**

```go
const (
	// IssuerConditionReasonInitializing is the value assigned to
	// the Reason field of the Ready condition when issuer-lib first
	// reconciles an Issuer which does not already have a Ready
	// condition.
	IssuerConditionReasonInitializing = "Initializing"
	IssuerConditionReasonPending = "Pending"
	IssuerConditionReasonChecked = "Checked"
	IssuerConditionReasonFailed = "Failed"
)
```

Reply:
The API does not match what we want to achieve eg. the IssuerStatus has this ACME field: https://github.com/cert-manager/cert-manager/blob/master/pkg/apis/certmanager/v1/types_issuer.go#L319C1-L333

Reply:
The values you listed are not present in the cert-manager/cert-manager repo I think ( I wasn't able to find them ).